modules/network/src/integrationTest/java/org/apache/ignite/internal/network/netty/ItConnectionManagerTest.java

```diff
@@ -302,57 +311,56 @@ public void testStopTwice() throws Exception {
-
-        assertThat(send2, willCompleteSuccessfully());
-        assertThat(send1.isDone(), is(true));
-
-        server1.close();
-        server2.close();
+            assertThat(send2, willCompleteSuccessfully());
+            assertTrue(send1.isDone());
+        }
     }
 
     @Test
     public void testTwoConnectionTypes() throws Exception {
-        int size = 1000000000;
-        char[] chars = new char[size];
-        Arrays.fill(chars, 'a');
-        String bigText = new String(chars);
-
+        String bigText = IgniteTestUtils.randomString(new Random(), 100000000);
 
-        ConnectionManagerWrapper server1 = startManager(4000);
-        ConnectionManagerWrapper server2 = startManager(4001);
+        Map<Integer, String> map = Map.of(1, bigText, 2, bigText);
 
-        NettySender sender1 = server1.openChannelTo(server2, ChannelType.DEFAULT).get(3, TimeUnit.SECONDS);
-        NettySender sender2 = server1.openChannelTo(server2, ChannelType.TEST).get(3, TimeUnit.SECONDS);
+        try (ConnectionManagerWrapper server1 = startManager(4000);
+                ConnectionManagerWrapper server2 = startManager(4001)) {
 
+            NettySender sender1 = server1.openChannelTo(server2).get(3, TimeUnit.SECONDS);
+            NettySender sender2 = server1.openChannelTo(server2, testChannel).get(3, TimeUnit.SECONDS);
 
-        TestMessage bigMessage = messageFactory.testMessage().msg(bigText).build();
-        TestMessage msg = messageFactory.testMessage().msg("test").build();
+            TestMessage bigMessage = messageFactory.testMessage().msg(bigText).map(map).build();
 
-        CompletableFuture<Void> send1 = sender1.send(new OutNetworkObject(bigMessage, Collections.emptyList()));
-        CompletableFuture<Void> send2 = sender2.send(new OutNetworkObject(msg, Collections.emptyList()));
+            CompletableFuture<Void> send1 = sender1.send(new OutNetworkObject(bigMessage, Collections.emptyList()));
+            CompletableFuture.runAsync(() -> {
+                for (int i = 0; i < 100; i++) {
+                    sender2.send(new OutNetworkObject(messageFactory.emptyMessage().build(), Collections.emptyList()));
+                }
+            });
 
-        assertThat(send2, willCompleteSuccessfully());
-        assertThat(send1.isDone(), is(false));
+            AtomicBoolean atLeastOneSmallWas = new AtomicBoolean(false);
+            server2.connectionManager.addListener(inNetworkObject -> {
+                System.out.println(inNetworkObject.message().groupType());
+                if (inNetworkObject.message().groupType() == EmptyMessageImpl.GROUP_TYPE
+                        && inNetworkObject.message().messageType() == EmptyMessageImpl.TYPE) {
+                    atLeastOneSmallWas.set(true);
+                }
+            });
 
-        server1.close();
-        server2.close();
+            assertThat(send1, willCompleteSuccessfully());
+            assertTrue(atLeastOneSmallWas.get());
```

Comment:
I don't understand these assertions. We check that by the time we received big message, at least one small message is received as well?
I think you could also add an assertion that be the time one of the small messages is received, big message is still sending.
Anyway, all of this still feels luck-based, was it really the only way to write the test? If there are no other options, I guess I'll approve it, but generally - I don't like this approach.
Another question - why have you decided to put these tests into "integrationTest" directory? Aren't these supposed to be unit tests?